Services/CustomDeviceModels.cs

```diff
@@ -67,7 +71,9 @@ public async Task<IEnumerable<DeviceModel>> GetListAsync()
             }
             catch (Exception e)
             {
-                this.log.Error("Unable to load device models from storage", e);
+                var msg = "Unable to load device models from storage";
+                this.log.Error(msg, e);
+                this.diagnosticsLogger.LogDiagnosticsData(SERVICE_ERROR_EVENT, $"{msg}: {e.Message}");
```

Comment:
I would suggest using a more semantic approach, e.g. instead of a generic `LogDiagnosticsData(..)` method, have a `LogServiceError(..)` method in the logger. This way you don't need to know which constant to use and the code is more readable. I would also delegate to the logger to extract information from the exception, passing the full object:

```
this.diagnosticsLogger.LogServiceError(msg, e)
```